File: leetcode2/0482. License Key Formatting.py
class Solution:
    def licenseKeyFormatting(self, S, K):
        """
        :type S: str
        :type K: int
        :rtype: str
        """
        S=S.replace('-','')
        S=S.upper()
        if not S:
            return ''
        firstpart=len(S)%K
        rep=[]
        if firstpart!=0:
            rep=[S[:firstpart]]+self.subsublicenseKeyFormatting(S[firstpart:],K)
        else:
            rep=rep+self.subsublicenseKeyFormatting(S,K)
        return '-'.join(rep)
    
    def subsublicenseKeyFormatting(self,S,K):
        if not S:
            return []
        rep=[]
        j=0
        while j<len(S):
            rep.append(S[j:j+K])
            j=j+K
        return rep

File: archives/C#/0482. License Key Formatting.cs
public class Solution {
    public string LicenseKeyFormatting(string S, int K) {
        S=S.Replace("-","");
        S=S.ToUpper();
        if(K>S.Length){
            return S;
        }
        int m=S.Length%K;
        if(m==0){
            return GetLicenseKeyFormatting(S,K);
        }
        else{
            return S.Substring(0,m)+"-"+GetLicenseKeyFormatting(S.Substring(m),K);
        }
    }
    public string GetLicenseKeyFormatting(string S,int K){
        string rep="";
        for(int i=0;i+K-1<S.Length;i+=K){
            rep=rep+"-"+S.Substring(i,K);
        }
        return rep.Substring(1);
    }
}
